src/metabase/legacy_mbql/normalize.cljc

```diff
@@ -399,12 +400,21 @@
   {:type            maybe-normalize-token
    ;; don't normalize native queries
    :native          normalize-native-query
-   :query           {:aggregation     normalize-ag-clause-tokens
-                     :expressions     normalize-expressions-tokens
-                     :order-by        normalize-order-by-tokens
-                     :source-query    normalize-source-query
-                     :source-metadata {::sequence normalize-source-metadata}
-                     :joins           {::sequence normalize-join}}
+   :query           {:aggregation        normalize-ag-clause-tokens
+                     :aggregation-idents (fn [m]
+                                           (update-keys m #(cond-> %
+                                                             (string? %) parse-long
+                                                             (keyword? %) (-> name parse-long))))
```

Comment:
Would this be too tricky?
```suggestion
                                                             (keyword? %) name
                                                             (string? %) parse-long)))
```

Reply:
A little... I'd rather keep them as separate clauses.

Reply:
Then maybe some kind of `cond` would be more explicit.